lti_placements/canvas_placements.py

```diff
@@ -245,9 +249,16 @@ def main() -> Sequence[DataSourceStatus]:
     '''
 
     canvas_env = ENV.get('CANVAS', {})
-    lti_processor = CanvasLtiPlacementProcessor(canvas_env.get("CANVAS_URL"), canvas_env.get("CANVAS_TOKEN"))
-    lti_processor.generate_lti_course_report(canvas_env.get("CANVAS_ACCOUNT_ID", 1), canvas_env.get(
-        "CANVAS_TERM_IDS", []), canvas_env.get("ADD_COURSE_IDS", []), True)
+    lti_processor = CanvasLtiPlacementProcessor(
+        canvas_env.get("CANVAS_URL"),
+        canvas_env.get("CANVAS_TOKEN"),
+        canvas_env.get("SUPPORTED_LTI_TOOLS", {}))
+
+    lti_processor.generate_lti_course_report(
+        canvas_env.get("CANVAS_ACCOUNT_ID", 1),
```

Comment:
looks like the only the root account will be checked for LTI tool report. I think we need to do recursive loop into all published courses in all accounts.

Maybe instead of using CanvasAPI's account.get_courses, we can use the course list reported in the course_inventory job?

Reply:
Yeah that's issue #162. Would require a larger refactor.

Reply:
Also I think that the other process is using the UM API, this is going direct to Canvas. This *might* be able to use the UM API now but I'm not sure. I wanted to support UM API as a backend to CanvasAPI but that is also some work.

Reply:
I don't quite understand what you mean, @zqian, by the "recursive loop into all published courses." We're not doing this in `COURSE_INVENTORY`. Wouldn't all the courses we care about be under the root account `1`, even if they have a specific sub-account?